shared_code/utils.py

```diff
@@ -145,3 +146,40 @@ def locate_potential_duplicate(name, zipcode, client, collection):
     if dupe_candidate is not None:
         return dupe_candidate["name"]
     return False
+
+
+def validate_data(df):
```

Comment:
validate_data() method  is currently not validating number 4 test case specified in github issue #58 . ie. To check if we have duplicate services for which serviceSummary can be combined into string of serviceSummaries separated by comma.

This is a very important validation because it will reduce the number of duplicates we add in temp collection if same resources is present in dataframe with different serviceSummary. Then in that case those two entries can be combined into single entry with serviceSummary field as comma separated values, hence reducing number of records in temp collection.

Reply:
I think that the `validate_data` function should not be used to alter the data frame.

I will create a new issue to deal with the combination of multiple entries into a single entry with the serviceSummary field as comma separated values. Ideally, this would be done in a separate function. What do you think?

Reply:
Yes, validate_data() method should only check if there exists entries which can be combined. The real job is done `aggregate_service_summary` method on base class but it depends on group_by_columns list (which we can actually default to be - [name, address1, city, state, and zip] ) 

I changed the issue you created to clearly specify not to implement it but rather just validate it. Which exactly is the job of validate_data method.